game/menu/gamedisplay.gd

```diff
@@ -1,12 +1,16 @@
+class_name GameDisplay
+
 extends MarginContainer
 
 signal pressed(game_file)
 
 var game_file: ConfigFile
-
+var _load_btn_main: Button
+var _load_btn_dialog: Button
```

Comment:
We load the load button dialog dynamically though. So on var would be quite annoying though and since that is just syntactic sugar anyway, i don't think it adds a lot of benefit for the additional mental costs.

Reply:
But the button is not dynamically loaded. It is loaded one time at the start.
That is exactly the purpose of onready vars.
`_load_btn_main` especially is not that hard because that button clearly never changes.
Btw. [official demos](https://github.com/godotengine/godot-demo-projects/blob/3.4-b0d4a7c/2d/platformer/src/Actors/Gun.gd) think the "benefit" to "additional mental cost" ratio is worth it.

Reply:
I found another [demo](https://github.com/godotengine/godot-demo-projects/blob/3.4-b0d4a7c/2d/tween/main.gd#L10-L21) that illustrates my point even better.
L21 even has a similar use case as _load_btn_dialog.

Reply:
_load_btn_dialog right now is created before _ready is called though:

![grafik](https://user-images.githubusercontent.com/833043/173065825-00ef399e-89e8-4bd0-a63c-83a248121bcd.png)

So my point stands. Not saying we can't do it like that. But it's not possible as is.

Reply:
More concretely:

You are either proposing to rewrite the current process in a major way (see also comment below)

Or the following, which would not work as expected right now.

![grafik](https://user-images.githubusercontent.com/833043/173066535-9e7fc8e6-ae2f-4a00-b87e-ff5251345438.png)

Reply:
It is not that much of a rewrite. 
Either delay the call of setup() with call_deferred or add the display to the scene tree immediately after instancing.
That is changing one line or moving it a few lines up respectively. Honestly, that is an oversight in my original PR.